File: java/EulerSolution.java
import java.util.concurrent.TimeUnit;

public interface EulerSolution {
    static void execute(EulerSolution solution) {
        long startTime = System.nanoTime();
        long result = solution.run();
        long endTime = System.nanoTime();

        long duration = TimeUnit.MILLISECONDS.convert(endTime - startTime, TimeUnit.NANOSECONDS);
        System.out.printf("Result: %d\n", result);
        System.out.printf("Time: %dms\n", duration);
    }

    long run();
}

File: kotlin/EulerSolution.kt
import java.util.concurrent.TimeUnit

interface EulerSolution {
    companion object {
        fun execute(soln : EulerSolution) {
            val start = System.nanoTime()
            val result = soln.run()
            var end = System.nanoTime()

            println("Result: $result")
            println("Result: ${TimeUnit.MILLISECONDS.convert(end - start, TimeUnit.NANOSECONDS)}ms")
        }
    }

    fun run() : Long
}